### ISSUE TITLE
Emissive/Unshaded output in CanvasItem shaders

### ISSUE THREAD

**Issue by andriyDev** [godotengine/godot-proposals#1478]
**Describe the project you are working on:**
2D overhead game with lighting effects.

**Describe the problem or limitation you are having in your project:**
CanvasItem shaders (at least VisualShaders) do not have an option to output "emissive"/unshaded colours. So you can either have only an unshaded Sprite, or a lit sprite, but not parts being lit and others unlit.

Example 1: A wolf with glowing red eyes. The wolf should be completely black when not in light (i.e. through a CanvasModulate), but it's eyes should be visible when in darkness.
Example 2: An enemy taking damage should flash red even when it darkness, but the enemy should not be visible in darkness.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
By having an emissive/unshaded output in CanvasItem shaders, this can simply be colour added to the output of the lighting pass.

Example 1: The wolf can have two sprite sheets, one for the actual wolf, and one for its eyes. Then the unshaded output simply samples from the second texture, meaning the eyes will still be visible even when in darkness.
Example 2: When taking damage, the emissive colour can be set to red, and once the damage has passed, the emissive colour can be set to transparent.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
Currently (by my understanding) the final output colour is just:
`final_colour = colour_from_light_pass`
The new code could be as simple as:
`final_colour = colour_from_light_pass + unshaded_from_fragment`
An even better approach would be to use the blending options. For example, mix:
final_colour = mix(colour_from_light_pass, unshaded_from_fragment, unshaded_from_fragment.a)

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
This feature can be worked around by having a second Sprite on top of the first, which is unlit and handles the emissive component, but this means twice as many nodes for each Sprite, and means you may need duplicated code in 2 separate shaders to get the same look.

**Is there a reason why this should be core and not an add-on in the asset library?:**
I don't think changes to the shader system work in an add-on. This seems to need to be implemented in C++ in the core engine rendering system.

**Comment by clayjohn** [comment]
Would it be okay to just have an EMISSION property the way that Spatial shaders do?

**Comment by andriyDev** [comment]
Yes, that's exactly what I'm looking for. I actually compiled my own version just for regular CanvasItem shaders, not VisualShaders. It was a 15 line change, but I still need to figure out how VisualShaders work to make a proper PR.

**Comment by Chaosus** [comment]
I could easily add it to the visual shaders if you commit to standard shaders first.

**Comment by Chaosus** [comment]
@andriyDev but if you really want to do it yourself - outputs are located at
https://github.com/godotengine/godot/blob/3b25548e4cef5f5a9292d1fb210d0bd66cbb407a/scene/resources/visual_shader.cpp#L2318
